hazelcast/src/main/java/com/hazelcast/config/WanPublisherState.java

```diff
@@ -0,0 +1,73 @@
+package com.hazelcast.config;
+
+/**
+ * Defines the state in which a WAN publisher can be in if it is not shutting
+ * down.
+ */
+public enum WanPublisherState {
+    /**
+     * State where both enqueuing new events is allowed, enqueued events are
+     * replicated to the target cluster and WAN sync is enabled.
+     */
+    ACTIVE((short) 0, true, true),
```

Comment:
Is there any background thought behind making the id field `short` and not `byte`? I'm fine with short, it doesn't really make difference, just out of curiosity.

Reply:
No, I first thought of `short` :) But it can be byte as well so I'll change it. I guess the whole effort with decreasing the memory consumption might be a moot point, might easily simply be an `int`.